### ISSUE TITLE
Accessibility: Clickable elements semantically not shown as such

### ISSUE THREAD

**Issue by fcnjd** [etkecc/synapse-admin#724]
**Describe the bug**
As a blind user working with screenreader and Braille display, HTML semantic is very important to navigate the UI. When e.g. in the users or rooms table, the entries are not tagged as a link or button, but anyway they have an option on them (assigned with JavaScript). Even though they might look like a button, if the semantic info is missing the assistive technology won't know.

**To Reproduce**
1. Go to the users page
2. Turn on a screenreader
3. Use the virtual cursor to find any user in the table
4. Find the screenreader not announcing button or link, but enter does something anyway

**Expected behavior**
An HTML button or a-element would be important here.

**Instance configuration:**
 - Synapse Admin version: v0.11.1-etke45
 - Synapse version not relevant

**Comment by beastafk** [comment]
Yes, it seems react-admin tests regularly for accessibility https://marmelab.com/react-admin/Features.html#accessibility 

And the mentioned component for the users (and also rooms, etc) page is [MUI React Table]( https://mui.com/material-ui/react-table/#accessibility). In the examples though, the table rows are not used as clickable elements  - so that might be the problem. 

Maybe we can add some customization like special `aria-` attributes to make screen readers happy?

**Comment by fcnjd** [comment]
Thank you for diving deeper into this problem. I've just verified, and you're right. The demo has the very same problem. It gets visible in two forms: When trying to select elements with tab (or arrow keys in the table), only the checkbox to toggle row selection is focusable, not the interactive element itself. Additionally, NVDA screenreader doesn't detect it as interactive, just as text. So yes, this demo behaves the same as Synapse-Admin here.

**Comment by aine-etke** [comment]
Unfortunately, there is no news about the issue from the react-admin side, so instead we implemented our own wrapper that attempts to improve the situation.

**Limitations**:
* Only desktop tables (datagrids) - mobile-friendly SimpleLists supposedly to have good accessibility by default
* Reference fields don't have proper titles - e.g., in room view -> members tab the list of users will use only user id as each col title due to nuances and complexity of the references implementation in react-admin
* Not all fields have human-readable values, e.g. dates and file sizes use raw values (i.e., unix timestamps and bytes)
* probably something else that was not identified yet

We hope this could work as a temporary workaround until proper accessibility implementation is done in react-admin

**Comment by fcnjd** [comment]
Thank you so much for your effort in this. I'm happy to hear that you implemented this change, and will test when the release is published.

**Comment by aine-etke** [comment]
you could try it even now using nightly builds: https://github.com/etkecc/ketesa?tab=readme-ov-file#-nightly-builds (of course, you waited almost a year already, so if a few more days or a week is not a big issue - no rush :)
